src/api/attestation_store.rs

```diff
@@ -48,11 +45,17 @@ pub const STORAGE_KEYS: &[usize] = &[1, 2];
 
 /// Implements a default attestation store using the environment store.
 ///
-/// The same attestation is used for batch and enterprise.
-pub trait Helper: Env {}
+/// Supports only one attestation at a time.
+pub trait Helper: Env {
+    /// Returns the current attestation id.
+    fn attestation_id(&self) -> Id;
```

Comment:
What the idea behind this function? That's only useful if an `Env` will have exactly one attestation type? And if you enable enterprise attestation, you have to remember to also call this function here to keep the state consistent? Doesn't that make the Helper unusable when we have a setup where the incoming MakeCredential parameters decide whether we use Enterprise or Batch?

Reply:
I think we decided that we don't support both batch and enterprise attestation simultaneously. If you want that, you need to implement the real trait yourself. This is just a helper. That's for simple usages, namely U2F and testing (i.e. stay backward compatible while enabling new usages).

Reply:
You can't use them at the same MakeCredential. But generally, the existance of enterprise attestation shouldn't affect attestation for all RP IDs that don't use enterprise attestation.

We can decide to not support batch attestation for enterprise enabled devices. That should then be a tested invariant in customization. Since individual attestation is preferred by OpenSK over batch attestation as the default customization setting, I think that's an okay shortcut to make.

Reply:
I'm not sure what you're talking about. There's 2 different things:
1. The `AttestationStore` API. This one supports batch and enterprise attestation simultaneously. That's the real API and that's what the CTAP library uses.
2. The `Helper` API. This one is just a helper to preserve the legacy behavior where only batch is supported (for U2F) except for tests where enterprise is also tested (although back then it was using the batch infrastructure since there was no separate notion of enterprise attestation).

If you're talking about 1, then I don't understand which issue you're referring about. If you're talking about 2, then I don't think we should support both simultaneously. That's just a legacy helper. Users who need fancy stuff should implement the real API.

Reply:
> So I wanted to check how useful this is for future us.

I don't think this is important. By definition of legacy, the `Helper` trait is for current users (or us, it's essentially the same thing). While the `AttestationStore` is for future users (or us).

> And suggest we include the customization invariant

What is this invariant?

Invariants should be specific to the CTAP library implementation, i.e. the assumptions that the CTAP library does over the environment. It shouldn't be invariants about the environnement implementations. Those are part of the tests of that environment implementation.

Reply:
Not sure what that means. We currently switch between EP and batch in tests. This is a necessary feature. It's not part of the `Helper` API so it has to be done by the implementer. The one for `TockEnv` doesn't while the one for `TestEnv` does.

Reply:
I inlined the `Helper` trait in `TockEnv` and `TestEnv`. Resolving.